src/sentry/utils/performance_issues/performance_detection.py

```diff
@@ -553,11 +564,186 @@ def visit_span(self, span: Span):
-                self.stored_issues[fingerprint] = PerformanceSpanIssue(
+                self.stored_issues[fingerprint] = PerformanceSpanProblem(
                     span_id, op_prefix, self.spans_involved[fingerprint]
                 )
 
 
+class NPlusOneDBSpanDetector(PerformanceDetector):
+    """
+    Detector goals:
+      - identify a database N+1 query with high accuracy
+      - collect enough information to create a good fingerprint (see below)
+      - only return issues with good fingerprints
+
+    A good fingerprint is one that gives us confidence that, if two fingerprints
+    match, then they correspond to the same issue location in code (and
+    therefore, the same fix).
+
+    To do this we look for a specific structure:
+
+      [-------- transaction span -----------]
+         [-------- parent span -----------]
+            [source query]
+                          [n0]
+                              [n1]
+                                  [n2]
+                                      ...
+
+    If we detect two different N+1 problems, and both have matching parents,
+    source queries, and repeated (n) queries, then we can be fairly confident
+    they are the same issue.
+    """
+
+    __slots__ = (
+        "stored_issues",
+        "potential_parents",
+        "source_span",
+        "n_hash",
+        "n_spans",
+    )
+
+    settings_key = DetectorType.N_PLUS_ONE_DB_QUERIES
+
+    def init(self):
+        self.stored_issues = {}
+        self.potential_parents = {}
+        self.n_spans = []
+        self.source_span = None
+
+    def visit_span(self, span: Span) -> None:
+        span_id = span.get("span_id", None)
+        op = span.get("op", None)
+        if not span_id or not op:
+            return
+
+        if op != "db":
+            # This breaks up the N+1 we're currently tracking.
+            self._maybe_store_issue()
+            self._reset_detection()
+            # Treat it as a potential parent as long as it isn't the root span.
+            if span.get("parent_span_id", None):
+                self.potential_parents[span_id] = span
+            return
+
+        if not self.source_span:
+            # We aren't currently tracking an N+1. Maybe this span triggers one!
+            self._maybe_use_as_source(span)
+            return
+
+        # If we got this far, we know we're a DB span and we're looking for a
+        # sequence of N identical DB spans.
+        if self._continues_n_plus_1(span):
+            self.n_spans.append(span)
+        else:
+            self._maybe_store_issue()
+            self._reset_detection()
+            # Maybe this DB span starts a whole new N+1!
+            self._maybe_use_as_source(span)
+
+    def visited_all_spans(self) -> None:
+        self._maybe_store_issue()
+
+    def _contains_complete_query(self, span: Span) -> bool:
+        # When relay truncates span description, it adds a "..." suffix (three
```

Comment:
is it relay or the sdk's after you've looked into it?

Reply:
Ah thanks it's the SDKs, will update comment

Reply:
Nice, good find 🔍 👏